include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php

```diff
@@ -0,0 +1,195 @@
+                    <td scope="row" >
+                        <input name="smbd_taxs" value="<?php echo $tax; ?>" type="radio" class="custom-tax">
+                    </td>
+                    <td>
+                        <label for="smbd_taxs"><?php echo $taxs[$tax]->labels->name; ?> </label>
+                    </td>
+                </tr>
+<?php
+			}
+?>
+            </table>
+
+            <h4><?php _e( 'The selected taxonomy has the following terms. Select the terms from which you want to delete posts', 'bulk-delete' ) ?></h4>
+            <p><?php _e( 'Note: The post count below for each term is the total number of posts in that term, irrespective of post type', 'bulk-delete' ); ?>.</p>
+<?php
+			foreach ( $terms_array as $tax => $terms ) {
+?>
+            <table class="optiontable terms_<?php echo $tax;?> terms">
+<?php
+				foreach ( $terms as $term ) {
+?>
+                    <tr>
+                        <td scope="row" >
+                            <input name="smbd_tax_terms[]" value="<?php echo $term->slug; ?>" type="checkbox" class="terms">
+                        </td>
+                        <td>
+                            <label for="smbd_tax_terms"><?php echo $term->name; ?> (<?php echo $term->count . ' '; _e( 'Posts', 'bulk-delete' ); ?>)</label>
+                        </td>
+                    </tr>
+<?php
+				}
+?>
+            </table>
+<?php
+			}
+?>
+            <table class="optiontable">
+				<?php
+				$this->render_filtering_table_header();
+				$this->render_restrict_settings();
+				$this->render_delete_settings();
+				$this->render_limit_settings();
+				$this->render_cron_settings();
+				?>
+            </table>
+
+            </fieldset>
+<?php
+			$this->render_submit_button();
+		} else {
+?>
+            <h4><?php _e( "This WordPress installation doesn't have any non-empty custom taxonomies defined", 'bulk-delete' ) ?></h4>
+<?php
+		}
+	}
+
+	protected function convert_user_input_to_options( $request, $options ) {
+		$options                       = array();
+		$options['post_type']          = bd_array_get( $request, 'smbd_tax_post_type', 'post' );
+		$options['selected_taxs']      = bd_array_get( $request, 'smbd_taxs' );
+		$options['selected_tax_terms'] = bd_array_get( $request, 'smbd_tax_terms' );
+		$options['restrict']           = bd_array_get_bool( $request, 'smbd_taxs_restrict', false );
+		$options['private']            = bd_array_get_bool( $request, 'smbd_taxs_private' );
+		$options['limit_to']           = absint( bd_array_get( $request, 'smbd_taxs_limit_to', 0 ) );
+		$options['force_delete']       = bd_array_get_bool( $request, 'smbd_taxs_force_delete', false );
+
+		$options['date_op']            = bd_array_get( $request, 'smbd_taxs_op' );
+		$options['days']               = absint( bd_array_get( $request, 'smbd_taxs_days' ) );
+
+		return $options;
+	}
+
+	public function delete( $delete_options ) {
+		if ( bd_array_get( $_POST, 'smbd_taxs_cron', 'false' ) == 'true' ) {
```

Comment:
You don't have to handle cron here. It is already handled. Check other metaboxes and also the `process_cron_filters` method.